src/neo/Ledger/Blockchain.cs

```diff
@@ -420,9 +423,16 @@ private void Persist(Block block)
                     ApplicationExecuted application_executed = new(engine);
                     Context.System.EventStream.Publish(application_executed);
                     all_application_executed.Add(application_executed);
+                    transactionStates[i] = new TransactionState
+                    {
+                        BlockIndex = block.Index,
+                        Transaction = tx,
+                        State = state
+                    };
                 }
                 using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.PostPersist, null, snapshot, block, system.Settings, 0))
                 {
+                    engine.SetState(transactionStates);
```

Comment:
@erikzhang Maybe we can remove block from `Create` and use a state?

Reply:
But many services rely on block.